File: 50-powx-n/powx-n.java
class Solution {
    public double myPow(double x, int n) {
        long N = n;

        // Edge case: 0 raised to negative power is undefined
        if (x == 0 && N < 0) {
            return Double.POSITIVE_INFINITY; // or throw exception
        }

        if (N < 0) {
            x = 1 / x;
            N = -N;
        }
        return power(x, N);
    }

    private double power(double x, long n) {
        if (n == 0) return 1.0;
        double half = power(x, n / 2);
        if (n % 2 == 0) {
            return half * half;
        } else {
            return half * half * x;
        }
    }
}

File: 50-powx-n/powx-n.c
double myPow(double x, int n) {
    double a=x,l;
    if(x==-1){
        if(n%2==0) return 1;
        else return -1;
    }
    if(x==1) return 1;
    if(n<-100000) return 0;
    if(n<0){
        a=1/x;
        for(int i=1;i<-n;i++){
            a=(1/x)*a;
        }
        return a;
    }
    else if(n==0) return 1;
    else{
        for(int i=1;i<n;i++){
        x=x*a;
        }
        return x;
    }
}